python/cudf/cudf/core/series.py

```diff
@@ -307,6 +307,82 @@ def append(self, other, ignore_index=False):
 
         return Series._concat([this, other], index=index)
 
+    def clip(self, lower=None, upper=None, inplace=False):
```

Comment:
Can we just move this entire function down to `Frame` and then handle it for DataFrame / Series / Index / MultiIndex?

Reply:
As issue with category has been taken out, I can try to get it under one hood.

Reply:
Added support to handle DataFrame and Series, pandas doesn't support clip for Index and MultiIndex.